Shopify/ruby-lsp
# Use fully qualified names in document symbol

### Motivation

Closes #113

Use the fully qualified names for constants and modules when displaying document symbols. In relation to the alternative way of solving this issue (displaying the values nested), I think this provides a better experience as it follows the flow of the file.

### Implementation

Created a helper method to find the fully qualified name of  a `ClassDeclaration` or `ModuleDeclaration` node. This recursively steps through each parent and builds the complete name.

### Automated Tests

Added a new fixture based on the example.

### Manual Tests

1. Open the LSP on this branch
2. Open the fixture `test/fixtures/one_line_nesting.rb`
3. Look at the file outline on the bottom left of VS Code
4. Verify that all classes/modules are using fully qualified names
